### PR TITLE
Beacons

### PR DESCRIPTION
**Pull request by sleepyyapril**
# Description

Changes carp rifts, nuke arming, and emergency shuttle to use readable locations.

I don't know EE codebase as well, so please request changes on anything I shouldn't have changed!

---

<details><summary><h1>Media</h1></summary>
<p>

Beacons example

</p>
</details>

---

# Changelog
nah

### CODE REVIEW

**Review by VMSolidus — APPROVED** [review]

**Review by Aidenkrz — APPROVED** [review]

File: Content.Server/Pinpointer/NavMapSystem.cs [hunk 1]
```diff
@@ -452,13 +476,11 @@ public string GetNearestBeaconString(MapCoordinates coordinates)
...
-        // we can null suppress the text being null because TryGetNearestVisibleStationBeacon always gives us a beacon with not-null text.
+        // we can null suppress the text being null because TRyGetNearestVisibleStationBeacon always gives us a beacon with not-null text.
         return Loc.GetString("nav-beacon-pos-format-direction",
             ("modifier", modifier),
             ("direction", ContentLocalizationManager.FormatDirection(adjustedDir).ToLowerInvariant()),
             ("color", beacon.Value.Comp.Color),
             ("marker", beacon.Value.Comp.Text!));
     }
-
-    #endregion
```
Comment (Aidenkrz) [inline Content.Server/Pinpointer/NavMapSystem.cs:463]: Don't remove #endregion, its whats throwing errors for you.

**Review by Aidenkrz — CHANGES_REQUESTED** [review]

**Comment by VMSolidus** [comment]
I'm unable to repair the Server NavMapSystem, since it looks like this was either a bad merge, and/or it is missing a dependency PR.

**Comment by sleepyyapril** [comment]
I can easily fix this, one moment.

File: Content.Server/Pinpointer/NavMapSystem.cs (hunk 1 above)
Comment (sleepyyapril) [reply Content.Server/Pinpointer/NavMapSystem.cs:463]: Damn, I didn't even realize I did that. Thanks!

File: Content.Server/Pinpointer/NavMapSystem.cs [hunk 2]
```diff
@@ -87,6 +90,27 @@ private void OnNavMapSplit(ref GridSplitEvent args)
         RefreshGrid(comp, _gridQuery.GetComponent(args.Grid));
     }
 
+    private void OnNavMapBeaconMapInit(EntityUid uid, NavMapBeaconComponent component, MapInitEvent args)
```
Comment (Aidenkrz) [inline Content.Server/Pinpointer/NavMapSystem.cs]: OnNavMapBeaconMapInit is defined twice, combine them into a single method.

File: Content.Server/Pinpointer/NavMapSystem.cs [hunk 3]
```diff
@@ -23,19 +28,17 @@ public sealed partial class NavMapSystem : SharedNavMapSystem
...
-    [Dependency] private readonly SharedMapSystem _mapSystem = default!;
-    [Dependency] private readonly SharedTransformSystem _transformSystem = default!;
+    [Dependency] private readonly TagSystem _tags = default!;
+    [Dependency] private readonly MapSystem _map = default!;
     [Dependency] private readonly IMapManager _mapManager = default!;
-    [Dependency] private readonly IGameTiming _gameTiming = default!;
-    [Dependency] private readonly ITileDefinitionManager _tileDefManager = default!;
+    [Dependency] private readonly TransformSystem _transform = default!;
+
+    private EntityQuery<PhysicsComponent> _physicsQuery;
```
Comment (Aidenkrz) [inline Content.Server/Pinpointer/NavMapSystem.cs:36]: your missing a using, Robust.Shared.Physics.Components;

**Review by Aidenkrz — CHANGES_REQUESTED** [review]

File: Content.Server/Pinpointer/NavMapSystem.cs (hunk 2 above)
Comment (sleepyyapril) [reply Content.Server/Pinpointer/NavMapSystem.cs]: No longer relevant

File: Content.Server/Pinpointer/NavMapSystem.cs (hunk 3 above)
Comment (sleepyyapril) [reply Content.Server/Pinpointer/NavMapSystem.cs:36]: No longer relevant

Comment (Aidenkrz) [reply Content.Server/Pinpointer/NavMapSystem.cs:36]: "no longer relevant" look at your checks

Comment (sleepyyapril) [reply Content.Server/Pinpointer/NavMapSystem.cs:36]: Oh, somehow relevant? I moved back to the commit with EE's stuff

**Pull request closed without merging**